### ISSUE TITLE
Accessibility and SEO audit for overlays with nested Navigation blocks

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#74583]
Part of https://github.com/WordPress/gutenberg/issues/73084, and a spin-off of https://github.com/WordPress/gutenberg/issues/73081.

We should audit and address any accessibility and SEO implications when Navigation overlays contain nested Navigation blocks or additional interactive content.

If there are multiple issues, we can use this issue as the parent issue.

**Comment by scruffian** [comment]
There was some useful input on this from the accessibility team:

> It's OK for the overlay to be separate in the HTML from the desktop version. Also, the navigation block when it's in the overlay should NOT be a nav . There should only be one wrapping nav on the overlay.

**Comment by getdave** [comment]
@jeryj I know you've been active in this area. If you had to summarise next steps here what would it be?

What's the must have vs nice to have?

**Comment by jeryj** [comment]
>  If you had to summarise next steps here what would it be?

What @scruffian reported :) 

> It's OK for the overlay to be separate in the HTML from the desktop version. Also, the navigation block when it's in the overlay should NOT be a nav . There should only be one wrapping nav on the overlay.

The overlay really needs to be wrapped by one <nav> item. The ideal is:

```
<nav>
  <button>Open Menu</button>
  <div class="overlay">
    <button>Close</button>
    <h2>Menu</h2>
    <ul>
      <li>navigation link 1</li>
      <li>navigation link 2</li>
    </ul>
  </div>
</nav>
```

This can be the same wrapping nav or different from the non-hamburger menu navigation. Either of these would be acceptable:

```
<!-- desktop view  -->
<nav>
 <ul>
      <li>navigation link 1</li>
      <li>navigation link 2</li>
    </ul>
</nav>
<!-- mobile view -->
<nav>
  <button>Open Menu</button>
  <div class="overlay">
    <button>Close</button>
    <h2>Menu</h2>
    <ul>
      <li>navigation link 1</li>
      <li>navigation link 2</li>
    </ul>
  </div>
</nav>
```
Or this:

```
<nav>
<!-- desktop view  -->
 <ul>
      <li>navigation link 1</li>
      <li>navigation link 2</li>
    </ul>
<!-- /desktop view  -->
<!-- mobile view  -->
 <button>Open Menu</button>
  <div class="overlay">
    <button>Close</button>
    <h2>Menu</h2>
    <ul>
      <li>navigation link 1</li>
      <li>navigation link 2</li>
    </ul>
  </div>
<!-- /mobile view  -->
</nav>

```

Conversation leading up to this HTML is from [the accessibility channel in Slack](https://wordpress.slack.com/archives/C02RP4X03/p1768323854903709).

**Comment by getdave** [comment]
Looks like I missed https://github.com/WordPress/gutenberg/pull/74764 which implemented the specific ask from the a11y team.

This is already done. Unless there are objections I'm going to close this Issue as it's too broad.
